### PR TITLE
Update package: Add eslint + Fix linter issues

### PR DESCRIPTION
**Pull request by susnux**
Some package cleanup:
* Add eslint + config
  * Add linting scripts
  * Fix all eslint issues
  * Add lint CI workflow
* Provide changelog in npm release
* Drop unneeded `.npmignore` -> already overridden by `files` section of `package.json`
* Update workflows

### CODE REVIEW

**Comment by ChristophWurst** [comment]
Mind splitting this up into reviewable chunks?

File: package.json [hunk 1]
```diff
@@ -40,23 +44,31 @@
...
   },
   "engines": {
     "node": "^20.0.0",
     "npm": "^9.0.0"
+  },
+  "eslintConfig": {
+    "extends": [
+      "@nextcloud/eslint-config/typescript"
+    ]
   }
```
Comment (skjnldsv) [inline package.json:73]: I prefer a dedicated file like all other repos :)

File: .github/workflows/node-test.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      - name: Set up npm ${{ steps.versions.outputs.npmVersion }}
+        run: npm i -g npm@"${{ steps.versions.outputs.npmVersion }}"
+
+      - name: Install dependencies & build
+        run: |
+          npm ci
+          npm run build --if-present
+
+      - name: Test
+        run: npm run test --if-present
```
Comment (skjnldsv) [inline .github/workflows/node-test.yml:47]: Wanna test types too?

**Review by skjnldsv — CHANGES_REQUESTED** [review]
Wow

**Comment by susnux** [comment]
Done see PRs

**Pull request closed without merging**